File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-11.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        f_map = {}
        maxLen = 0
        l = r = 0
        maxFreq = 0
        while r < len(s):
            f_map[s[r]] = f_map.get(s[r],0) + 1
            maxFreq = max(maxFreq,f_map[s[r]])

            if r-l+1 - maxFreq > k:
                f_map[s[l]] -= 1
                l += 1
            
            maxLen = max(maxLen, (r-l+1))
            r += 1
        return maxLen

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-11.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> numSet(nums.begin(), nums.end());
        int maxLen = 0;
        for(auto num: numSet){
            if(!numSet.count(num - 1)){
                int length = 1;
                int nextNum = num + 1;
                while(numSet.count(nextNum)){
                    length += 1;
                    nextNum += 1;
                }
                maxLen = max(maxLen, length);
            }
        }
        return maxLen;
    }
};
